### PR TITLE
feat(performance-metrics): Add metrics series data widget [INGEST-680]

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Port singleFieldAres widget to metrics -  p75 LCP
     - [x] p75 LCP
     - [x] TPM
     - [ ] User Misery (it's part of another story)
 - [x] Write Tests
     - [x] p75 LCP
     - [ ] TPM (will do in another PR as this one is getting pretty long)
     - [ ] User Misery (it's part of another story)
- [x] Add `includePrevious` to the MetricsRequest component
- [x] moved `shouldFetchPreviousPeriod` function from projectDetails to `charts/utils `

### CODE REVIEW

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        }),
+      };
+    }) ?? [];
+
+  const dataMean = data.map(series => {
+    const meanData = mean(series.data.map(({value}) => value));
+    return {
+      mean: meanData,
+      outputType: aggregateOutputType(series.seriesName),
+      label: axisLabelFormatter(meanData, series.seriesName),
```
Comment (matejminar) [inline static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:41]: We need to make sure that the output of these functions is correct for all fields.
Not sure if our naming is 1:1 with all transaction fields (tpm and failure_rate come to my mind).

**Review by matejminar — COMMENTED** [review]
So far looks good, left a couple of early-feedback suggestions.

Let's not forget to add test coverage in `tests/js/spec/views/performance/landing/widgets/widgetContainer.spec.jsx`

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:41]: sure... I might have to update it when implementing tpm

File: static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidgetMetrics.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  const globalSelection = eventView.getGlobalSelection();
+
+  if (fields.length !== 1) {
+    throw new Error(`Single field area can only accept a single field (${fields})`);
+  }
+
+  const field = fields[0];
+
+  // TODO(metrics): make this list complete once api is ready
+  const metricsFieldMap = {
```
Comment (priscilawebdev) [inline static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidgetMetrics.tsx:47]: still want to improve this. Glad to get feedback :)

**Review by k-fish — APPROVED** [review]
Didn't look as closely at `metricsRequest` etc., but all the performance widget code looks good 👍

File: static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidgetMetrics.tsx (hunk 2 above)
Comment (k-fish) [reply static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidgetMetrics.tsx:47]: Yeah field map might not be ideal but it works, the only other thing you could do is modify the `widgetDefinitions` to have `metricFields` and only replace it with field, but I honestly prefer leaving all the split code inside the widgets, as there are already a couple branches for metrics already. 🤔

File: static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx:41]: it works for tpm https://github.com/getsentry/sentry/blob/efc144f6069dcf510122fddb5c3906a9bd642eeb/static/app/utils/discover/fields.tsx#L117

**Review by matejminar — APPROVED** [review]

**Pull request merged**